src/Service.GraphQLBuilder/Mutations/CreateMutationBuilder.cs

```diff
@@ -192,12 +248,41 @@ private static InputValueDefinitionNode GenerateSimpleInputType(NameNode name, F
                 location: null,
                 field.Name,
                 new StringValueNode($"Input for field {field.Name} on type {inputTypeName}"),
-                type,
+                databaseType is DatabaseType.MSSQL ? type.NullableType() : type,
                 defaultValue: null,
-                field.Directives
+                databaseType is DatabaseType.MSSQL ? new List<DirectiveNode>() : field.Directives
             );
         }
 
+        /// <summary>
+        /// Helper method to determine if there is a M:N relationship between the parent and child node.
+        /// </summary>
+        /// <param name="fieldDefinitionNode">FieldDefinition of the child node.</param>
+        /// <param name="childObjectTypeDefinitionNode">Object definition of the child node.</param>
+        /// <param name="parentNode">Parent node's NameNode.</param>
+        /// <returns></returns>
+        private static bool IsMToNRelationship(FieldDefinitionNode fieldDefinitionNode, ObjectTypeDefinitionNode childObjectTypeDefinitionNode, NameNode parentNode)
+        {
+            Cardinality rightCardinality = RelationshipDirectiveType.Cardinality(fieldDefinitionNode);
+            if (rightCardinality is not Cardinality.Many)
+            {
+                return false;
+            }
+
+            List<FieldDefinitionNode> fieldsInChildNode = childObjectTypeDefinitionNode.Fields.ToList();
+            int index = fieldsInChildNode.FindIndex(field => field.Type.NamedType().Name.Value.Equals(QueryBuilder.GeneratePaginationTypeName(parentNode.Value)));
```

Comment:
How do we guarantee the child node has been processed completely such that all its relationship fields including the parent are already added and we find the parent here? for e.g. in case of book -> authors, how can we guarantee complex type for author has already been created so as to find the book (parent) under author (child)?

Reply:
I think there is some confusion.Here, we are only talking about the ObjectTypeDefinition(s)/FieldDefinition(s) which are created before hand. What we are doing here is we are creating InputObjectTypeDefinition(s) using the already created ObjectTypeDefinition(s)/FieldDefinition(s).